homeassistant/components/overkiz/coordinator.py

```diff
@@ -209,3 +209,32 @@ async def on_execution_state_changed(
         ExecutionState.FAILED,
     ]:
         del coordinator.executions[event.exec_id]
+
+
+class OverkizDeviceRefreshCoordinator(DataUpdateCoordinator[None]):
+    """Class to trigger device state refresh for devices on Overkiz platform."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        *,
+        client: OverkizClient,
+        device_url: str,
+    ) -> None:
+        """Initialize data update coordinator."""
+        self.client = client
+        self.device_url = device_url
+        super().__init__(
+            hass,
+            logger,
+            name=f"device refresh for {device_url}",
+            update_interval=REFRESH_DEVICE_STATES_INTERVAL,
+        )
+
+        # Ensure this coordinator runs periodically even though there is no listener needed.
+        self.async_add_listener(lambda: None)
+
+    async def _async_update_data(self) -> None:
+        """Trigger device state refresh on Overkiz platform."""
+        await self.client.refresh_device_states(self.device_url)
```

Comment:
This is not really a use-case where the data update coordinator is required imo, since you don't receive any data here.

See https://github.com/iMicknl/ha-tahoma/pull/271 for my initial try 3 years ago. Code is very outdated, but shows you another approach. You could use `async_track_time_interval` to call a function every x minutes.

Reply:
Allright, I'll migrate it to `async_track_time_interval`.

Reply:
Great, let me know if you need help! This issue (https://github.com/home-assistant/core/issues/103183) looks related.